### PR TITLE
Adding a Spinner slot to the blankslate component

### PR DESCRIPTION
**Pull request by jonrohan**
This addresses a need from @colebemis and I, we're replacing `gif` spinners on github.com with the new SpinnerComponent. We came across a couple of uses where the `image_src` property was being used to display a loading image in the blankslate component. 

This PR adds a `spinner` slot that passes in the SpinnerComponent directly.

image

### CODE REVIEW

File: app/components/primer/blankslate_component.rb [hunk 1]
```diff
@@ -17,6 +21,14 @@ class BlankslateComponent < Primer::Component
     #     description: "Description",
     #   ) %>
     #
+    # @example 150|Loading|Add a [SpinnerComponent](https://primer.style/view-components/components/spinner) to the blankslate in place of an icon.
+    #   <%= render Primer::BlankslateComponent.new(
+    #     title: "Title",
+    #     description: "Description",
+    #   ) do |component| %>
+    #     <% component.slot(:spinner) %>
```
Comment (joelhawksley) [inline app/components/primer/blankslate_component.rb:29]: Might we provide an example of an argument passed to the spinner, such as the size?

File: app/components/primer/blankslate_component.rb [hunk 2]
```diff
@@ -112,5 +124,17 @@ def initialize(
       @link_text = link_text
       @link_url = link_url
     end
+
+    class Spinner < Primer::Slot
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
```
Comment (joelhawksley) [inline app/components/primer/blankslate_component.rb:129]: Might we include the arguments for SpinnerComponent as well?

File: app/components/primer/blankslate_component.html.erb [hunk 3]
```diff
@@ -1,13 +1,13 @@
 <%= render Primer::BaseComponent.new(**@system_arguments) do %>
-  <% if @icon.present? %>
+  <% if spinner.present? %>
+    <%= render spinner.component %>
+  <% elsif @icon.present? %>
```
Comment (jonrohan) [inline app/components/primer/blankslate_component.html.erb:4]: One small change here @joelhawksley, which I think is a bug fix. You can't have a `spinner`, `@icon`, and a `@image_src` at the same time.

**Review by joelhawksley — APPROVED** [review]
Looks great! Mind adding a CHANGELOG entry? We can then ship a release with this change.

File: app/components/primer/blankslate_component.html.erb (hunk 3 above)
Comment (joelhawksley) [reply app/components/primer/blankslate_component.html.erb:4]: 👍🏻  I like it.

**Pull request merged**